### PR TITLE
Travis build configuration with heroku deployment

### PR DESCRIPTION
**Pull request by vinaymavi**
Pull request contains code for issues #13 #29

### CODE REVIEW

**Comment by vinaymavi** [comment]
Travis heroku deployment guide - https://docs.travis-ci.com/user/deployment/heroku/

**Comment by areai51** [comment]
Hey  @vinaymavi Travis Config and Heroku deployment is a great PR. thanks a ton for this. @sbmadhav @vichu1988  can either of you do a quick review of this so that I can merge it to master before it gets too late.

File: .travis.yml [hunk 1]
```diff
@@ -1 +1,8 @@
-language: node_js
\ No newline at end of file
+language: node_js
+node_js:
+- iojs
+deploy:
+  provider: heroku
+  api_key:
+    secure: SSnj6mv6oJ/riKhkcGFG3pCqkobkd20VODhYIbAHE6eYDUZg7onOI00s7Aj7TZkvyJEeudVRdplNDFgAy5AFeVq6MbdQgkXgBST4XWXExF5rneZ06YqwKOokCARYfmx2Kw+25yiI87AJkY2CuS909kOFDk7VbbnJIu8McBLDTx+K076QrOkZFZWtSdx+30ndvTH0hgPPaf69ze5l4tKO+3cFASPiqnXZxTzTFxEhhgy3lw1hd0f1M5ZtCcnUQklIsIGam4p9X98QusYyPcmw/1SzVjSbLWHLUIWygQvGr2zIAzRzAEQG44xn8CHqGWki+aRK6XLqOqNaQt6MY4b9DhnFleuHUo+n9pL5PZ9fz323MIZJ1uD6VO1OO9Nc8qb4z3IYxsDhzTnj2h8KCnw64FcmosxbeSTbBjUMzGvnNbZdTgU3r2o9OzMG72SBCzmlD0pIR3e9LLzBTY/SiT1jld1vsFm2LceNSz6TZZeVViOPQ5bqbiKtbYn0bqL3Ja8UCsZXumejL7QEJo6fzCVBgrIuH7+SKC4R3/bVsspbqwRVHX+Y2oTGzwBId33FiAwXfWfLIUVxycpHOormdkRGyExho+4zXs0US+CUoIZyfikTDpx1FIbwR/PCFytxtPxAhnEhUzakhCFpNK1umgJ0cQpgyvzuPDRgac+AKbxhpt4=
```
Comment (sbmadhav) [inline .travis.yml]: We should not commit the API Key. Is it possible to move this to runtime config?

File: package.json [hunk 2]
```diff
@@ -28,14 +28,14 @@
...
     "build": "next build",
     "prestart": "npm run clean && npm run build",
-    "start": "next start -p $PORT",
+    "start": "next start",
+    "heroku-start": "npm run clean && npm run build && next start -p $PORT",
     "clean": "rimraf .next",
     "lint": "eslint pages fusion",
-    "pre-commit": "npm run lint"
+    "pre-commit": "npm run lint",
     "heroku-postbuild": "next build"
```
Comment (sbmadhav) [inline package.json]: Can this use the same convention as others to 
 `"heroku-postbuild": "npm run build"`
That way not many changes if we plan to move out next later on

**Review by sbmadhav — CHANGES_REQUESTED** [review]

**Comment by vish288** [comment]
Add CI pill onto readme
and also update Read me to point to demo site.

File: .travis.yml (hunk 1 above)
Comment (vinaymavi) [reply .travis.yml]: @sbmadhav  rung config is a great idea, i need to check how to implement it. 
This api key is not the original one, it is a hash that generated by **travis CLI**.
can we continue current approach? if no, please provide some pointer how we can implement **run config**?

File: package.json (hunk 2 above)
Comment (vinaymavi) [reply package.json]: done.

File: .travis.yml (hunk 1 above)
Comment (sbmadhav) [reply .travis.yml]: If API key is a hash that generated by Travis CLI, would that mean that someone else with the key can not connect to the same CLI? Is there another password that you use to connect?

**Comment by vinaymavi** [comment]
@vichu1988 CI pill and demo site link updated, can we add demo link in website section also?
image

File: .travis.yml (hunk 1 above)
Comment (vinaymavi) [reply .travis.yml]: I am not sure @sbmadhav, just followed the document -  https://docs.travis-ci.com/user/deployment/heroku/ and it suggest this method.

Comment (vish288) [reply .travis.yml]: easier way would be enable it via owners and co-owners. @areai51 : Lets take out api key from the config and switch on travis CI build via owner integration of the repo. Could you do that pls?

File: package.json [hunk 3]
```diff
@@ -28,15 +28,15 @@
     "theming": "^1.1.0"
   },
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
```
Comment (vish288) [inline package.json:31]: point test to as of now run lint alone. any component we touch , lets ensure we clean up any and all linting errors.

**Review by sbmadhav — APPROVED** [review]

File: .travis.yml (hunk 1 above)
Comment (sbmadhav) [reply .travis.yml]: Ah! okay. As long as it is the public key we are good. I assume you have run this step correct ?
`travis encrypt $(heroku auth:token) --add deploy.api_key`

**Comment by sbmadhav** [comment]
@areai51  - we are good to merge this pull request now.

File: .travis.yml (hunk 1 above)
Comment (vinaymavi) [reply .travis.yml]: yes, i ran same command.

**Review by vish288 — COMMENTED** [review]
Please change 
`
"test": "echo \"Error: no test specified\" && exit 1",
`
to 
`
"test": "npm run lint",
`

**Pull request merged**